authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,734 @@
++
+[source,terminal]
+----
+$ ./sts-preflight token
+----
++
+This token is used to assume a role to get the credentials that are required to start the installation. The token is created in the `_output/token` directory.
+
+. Use the token to assume the installer role and mint the credentials that are required for the {product-title} installation process.
++
+[source,terminal]
+----
+$ ./sts-preflight assume
+----
++
+The output prints several `export` commands.
+
+. Run the `export` commands from the output of the previous step.
++
+[source,terminal]
+----
+$ export AWS_ACCESS_KEY_ID=<aws-access-key-id>
+----
++
+[source,terminal]
+----
+$ export AWS_SECRET_ACCESS_KEY=<aws-secret-access_>
+----
++
+[source,terminal]
+----
+$ export AWS_SESSION_TOKEN=<aws-session-token>
+----
+
+[id="sts-mode-installing-preflight-run-installer"]
+==== Running the installer
+
+. Run the {product-title} installer
++
+[source,terminal]
+----
+$ ./openshift-install create cluster --log-level=debug
+----
+
+. Continue to xref:#sts-mode-installing-verifying[Verifying the installation].
+
+[id="sts-mode-installing-manual-config"]
+=== Configuring a cluster manually
+
+When xref:#sts-mode-installing[installing] an {product-title} cluster that is configured to use the CCO in manual mode with STS, you can manually create the required AWS resources.
+
+As an alternative, you can use the xref:#sts-mode-installing-preflight-config[`sts-preflight` tool] to create AWS resources.
+
+.Procedure
+
+. Generate a private key to sign the `ServiceAccount` object:
++
+[source,terminal]
+----
+$ openssl genrsa -out sa-signer 4096
+----
+
+. Generate a `ServiceAccount` object public key:
++
+[source,terminal]
+----
+$ openssl rsa -in sa-signer -pubout -out sa-signer.pub
+----
+
+. Create an S3 bucket to hold the OIDC configuration:
++
+[source,terminal]
+----
+$ aws s3api create-bucket --bucket <oidc-bucket-name> --region <aws-region>
+----
+
+. Retain the S3 bucket URL:
++
+[source,terminal]
+----
+OPENID_BUCKET_URL="https://s3.<aws-region>.amazonaws.com/<oidc-bucket-name>
+----
+
+. Build an OIDC configuration:
+
+.. Save a file named `keys.json` that contains the following information:
++
+[source,json]
+----
+{
+    "keys": [
+        {
+            "use": "sig",
+            "kty": "RSA",
+            "kid": "<public-signing-key-id>",
+            "alg": "RS256",
+            "n": "<public-signing-key-modulus>",
+            "e": "<public-signing-key-exponent>"
+        }
+    ]
+}
+----
++
+Where:
+
+*** `<public-signing-key-id>` is generated from the public key with:
++
+[source,terminal]
+----
+$ openssl rsa -in sa-signer.pub -pubin --outform DER | openssl dgst -binary -sha256 | openssl base64 | tr '/+' '_-' | tr -d '='
+----
++
+This command converts the public key to DER format, performs a SHA-256 checksum on the binary representation, encodes the data with base64 encoding, and then changes base64-encoded output to use only characters that can be used in a URL.
```

Comment:
s/and then changes base64-encoded output/and then changes the base64-encoded output... ??

Or;
and changes the base64-encoded output... ??

Reply:
Same in all 3 bullets here